setup.py

```diff
@@ -125,129 +125,182 @@ def clean_line(line):
     __version__ = version_file.read().strip()
 
 
-# Cython extensions:
-def get_ext_modules():
+# %% Cython extensions:
+# Ensures RADIS still builds if without Cython
+
+
+class BuildFailed(Exception):
+    pass
+
+
+from distutils.errors import CCompilerError, DistutilsExecError, DistutilsPlatformError
+
+
+def show_message(*lines):
+    """ Note : will only happen if user installs with `pip install -v` """
+    print("=" * 74, file=sys.stderr)
+    for line in lines:
+        print(line, file=sys.stderr)
+    print("=" * 74, file=sys.stderr)
+
+
+def get_ext_modules(with_binaries):
+    """
+    Parameter
+    ---------
+    with_binaries: bool
+        if False, do not try to build Cython extensions
+    """
+    # Based on code from https://github.com/pallets/markupsafe/blob/main/setup.py
+
     print(sys.version)
     ext_modules = []
     cmdclass = {}
 
+    if not with_binaries:
+        # skip building extensions
+        return {"cmdclass": cmdclass, "ext_modules": ext_modules}
+
     # TO-DO: set language level
 
     try:
         import cython
-
-        print("Cython " + cython.__version__)
-
-        from Cython.Distutils import build_ext
-
-        class build_ext_subclass(build_ext):
-            def build_extensions(self):
-                c = self.compiler.compiler_type
-                copt = {
-                    "msvc": ["/openmp", "/Ox", "/fp:fast", "/favor:INTEL64"],
-                    "mingw32": ["-fopenmp", "-O3", "-ffast-math", "-march=native"],
-                }
-
-                lopt = {"mingw32": ["-fopenmp"]}
-
-                print("Compiling with " + c + "...")
-                try:
-                    for e in self.extensions:
-                        e.extra_compile_args = copt[c]
-                except (KeyError):
-                    pass
-                try:
-                    for e in self.extensions:
-                        e.extra_link_args = lopt[c]
-                except (KeyError):
-                    pass
+    except (ModuleNotFoundError) as err:
+        raise BuildFailed(
+            "Cython not found : Skipping all Cython extensions...!"
```

Comment:
At some point we should pack the .cpp files so that the absence of the cython package doesn't preclude compilation of the c/c++ code. But it's fine the way it is for now!